### PR TITLE
Fix oas3 definition for GA

### LINKED ISSUE #3596: v3 version of GetTransactionByHash is broken in v0.0.0-iris
I have created a generalized account:
```
$ curl http://localhost:3013/v3/accounts/ak_zgD1GRzKrJfQJzSvZDqkFgzgbH2ZWuv7RBLbUcYHsRtGwqmXQ
{"auth_fun":"authorize","balance":99999920704000000000,"contract_id":"ct_KGjzz9EvuQx3JBgPcczuTWwbtszwhvZssUdfCWeSYZmKwrgkA","id":"ak_zgD1GRzKrJfQJzSvZDqkFgzgbH2ZWuv7RBLbUcYHsRtGwqmXQ","kind":"generalized","nonce":1,"payable":true}%
```

When I'm getting the details of the corresponding transaction using v2 endpoint it works fine:
```
$ curl http://localhost:3013/v2/transactions/th_BizDRu3neztChKpKBwRuxFHtx4YsQ5qjN5MfwT8Xo4j7wtmi3
{"block_hash":"mh_2by8G636JbtYVfvqU8b31M4SiJwC6HTuWmWrAV81KpKbANBgwp","block_height":697,"hash":"th_BizDRu3neztChKpKBwRuxFHtx4YsQ5qjN5MfwT8Xo4j7wtmi3","signatures":["sg_G6fNN6RAgpkFsd7kStNrDG5o5cfJisdNSDmj5QD6c8229yGRuCKzLCESERcGtmwTnoErYiAXat8TfAEEE19noANQiYYCu"],"tx":{"abi_version":3,"auth_fun":"0x6cf2570b0a1599b708291e50aa3daf13d0c7f2484bc337ddad2413a37fd4a009","call_data":"cb_KxFE1kQfP4oEp9E=","code":"cb_+GtGA6BmBAe/FJsLKdVmOGcgSYtf1zC0+5KbEImjxDkv0YxR28C4Pp7+RNZEHwA3ADcAGg6CPwEDP/5s8lcLADcBBxcBA/+bLwIRRNZEHxFpbml0EWzyVwslYXV0aG9yaXplgi8AhTQuMy4wAPuOiKw=","fee":79240000000000,"gas":1579000,"gas_price":1000000000,"nonce":1,"owner_id":"ak_zgD1GRzKrJfQJzSvZDqkFgzgbH2ZWuv7RBLbUcYHsRtGwqmXQ","type":"GAAttachTx","version":1,"vm_version":5}}%
```

But if I'm using v3 endpoint it fails with 500 code:
```
$ wget http://localhost:3013/v3/transactions/th_BizDRu3neztChKpKBwRuxFHtx4YsQ5qjN5MfwT8Xo4j7wtmi3
--2021-04-27 16:32:58--  http://localhost:3013/v3/transactions/th_BizDRu3neztChKpKBwRuxFHtx4YsQ5qjN5MfwT8Xo4j7wtmi3
Resolving localhost (localhost)... ::1, 127.0.0.1
Connecting to localhost (localhost)|::1|:3013... connected.
HTTP request sent, awaiting response... 500 Internal Server Error
2021-04-27 16:32:58 ERROR 500: Internal Server Error.
```
at the same time in node's logs appears:
```
node_1      | 13:32:58.566 [error] CRASH REPORT Process <0.20003.1> with 0 neighbours crashed with reason: [{data_invalid,[{<<"type">>,<<"object">>},{<<"discriminator">>,[{<<"propertyName">>,<<"type">>}]},{<<"properties">>,[{<<"version">>,[{<<"$ref">>,<<"#/components/schemas/UInt32">>}]},{<<"type">>,[{<<"type">>,<<"string">>},{<<"enum">>,[<<"SpendTx">>,<<"ChannelCreateTx">>,<<"ChannelDepositTx">>,<<"ChannelWithdrawTx">>,<<"ChannelForceProgressTx">>,<<"ChannelCloseMutualTx">>,<<"ChannelCloseSoloTx">>,<<"ChannelSlashTx">>,<<"ChannelSettleTx">>,<<"ChannelSnapshotSoloTx">>,<<"ChannelSetDelegatesTx">>,...]}]}]},...],...}] in jesse_error:default_error_handler/3 line 78
node_1      | 13:32:58.650 [error] Cowboy stream 1 with ranch listener external and connection process <0.20002.1> had its request process exit with reason: {nocatch,[{data_invalid,[{<<"type">>,<<"object">>},{<<"discriminator">>,[{<<"propertyName">>,<<"type">>}]},{<<"properties">>,[{<<"version">>,[{<<"$ref">>,<<"#/components/schemas/UInt32">>}]},{<<"type">>,[{<<"type">>,<<"string">>},{<<"enum">>,[<<"SpendTx">>,<<"ChannelCreateTx">>,<<"ChannelDepositTx">>,<<"ChannelWithdrawTx">>,<<"ChannelForceProgressTx">>,<<"ChannelCloseMutualTx">>,<<"ChannelCloseSoloTx">>,<<"ChannelSlashTx">>,<<"ChannelSettleTx">>,<<"ChannelSnapshotSoloTx">>,<<"ChannelSetDelega...">>,...]}]}]},...],...}]} in jesse_error:default_error_handler/3 line 78
```

### PR DESCRIPTION
**Pull request by velzevur**
Fixes #3596

The issue was that with the new `oas3` we do check the pattern of the auth
function for `ga_attach_tx`. We don't do that for `swagger2` API. Since we didn't have integration tests coverage, this slipped in. Now we have tests to cover attach and meta transactions.

The work on this PR had been supported by the ACF.

### CODE REVIEW

**Review by cytadela8 — DISMISSED** [review]

**Review by hanssv — APPROVED** [review]

File: apps/aehttp/src/aehttp_dispatch_ext.erl [hunk 1]
```diff
@@ -428,7 +428,9 @@ handle_request_('GetTransactionByHash', Params, _Config) ->
                     {200, [], SerializedTx};
                 {BlockHash, Tx} ->
                     {ok, Header} = aec_chain:get_header(BlockHash),
-                    {200, [], aetx_sign:serialize_for_client(Header, Tx)}
+                    Response = aetx_sign:serialize_for_client(Header, Tx),
+                    lager:debug("ASDF ~p", [Response]),
```
Comment (hanssv) [inline apps/aehttp/src/aehttp_dispatch_ext.erl:432]: ASDF is perhaps not the most helpful debug log message?

**Review by hanssv — APPROVED** [review]

**Review by hanssv — APPROVED** [review]

File: apps/aehttp/priv/oas3.yaml [hunk 2]
```diff
@@ -2480,8 +2480,6 @@ components:
           $ref: "#/components/schemas/EncodedPubkey"
         responder_amount:
           $ref: "#/components/schemas/UInt"
-        push_amount:
```
Comment (velzevur) [inline apps/aehttp/priv/oas3.yaml:2483]: This is not present in the transaction

File: apps/aehttp/priv/oas3.yaml [hunk 3]
```diff
@@ -2891,7 +2888,6 @@ components:
             $ref: "#/components/schemas/EncodedValue"
       required:
         - tx
-        - signatures
```
Comment (velzevur) [inline apps/aehttp/priv/oas3.yaml:2894]: Meta signed txs do not have signatures

File: apps/aehttp/priv/oas3.yaml [hunk 4]
```diff
@@ -3457,7 +3453,7 @@ components:
         auth_fun:
           description: Contract authorization function hash (hex encoded)
           type: string
-          pattern: ^(0x|0X)?[a-fA-F0-9]+$'
```
Comment (velzevur) [inline apps/aehttp/priv/oas3.yaml:3460]: Typo

File: apps/aehttp/src/aehttp_api_validate.erl [hunk 5]
```diff
@@ -45,7 +45,7 @@ validator(SpecVsn) ->
 
 -spec response(
     OperationId :: atom(),
-    Methohd :: binary(),
```
Comment (velzevur) [inline apps/aehttp/src/aehttp_api_validate.erl:48]: Typo

**Review by uwiger — APPROVED** [review]

**Pull request merged**